lib/util/source-code.js

```diff
@@ -114,6 +115,12 @@ function SourceCode(text, ast) {
      */
     this.lines = text.split(/\r\n|\r|\n|\u2028|\u2029/g);
 
+    /**
+     * Fixes to be made later on.
+     * @type Object[]
+     */
+    this._fixes = [];
```

Comment:
This implementation is naive as a first pass, I need to put some restrictions on it so that you can't make two changes to the same area. That will help a bit.

> if one rule fixes an issue, which in turn will cause another rule to fail, failure will not be reported

This is where we need to be sure we don't have rules that overlap. In theory, if we have completely atomic rules, we shouldn't run into this case. However, we do need to be careful about which rules will do fixes and which will not. If we run into a case where two fixes collide, then we'll need to decide which fix to turn off. That's why I'm starting with just a small group of rules rather than trying to do it for every rule we have right off the bat. We can release them in small batches to see if we run into any problems.

Ultimately, I'd like to figure out a syntax tree approach to doing this, but that's a much more difficult problem given our single-traversal model. So, I see this as an intermediate step to get some quick wins before really digging into that.